polyfill/test/duration.mjs

```diff
@@ -1065,6 +1065,147 @@ describe('Duration', () => {
+      const oneMonth = Duration.from({ months: 1 });
+      equal(oneMonth.total({ unit: 'months', relativeTo: { year: 2020, month: 1, day: 1, months: 2 } }), 1);
+    });
+    it('throws if unit is missing', () => {
+      [undefined, {}, () => {}, { roundingMode: 'ceil' }].forEach((options) =>
+        throws(() => d.total(options), RangeError)
+      );
+    });
+    it('relativeTo is required for rounding calendar units even in durations without calendar units', () => {
+      throws(() => d2.total({ unit: 'years' }), RangeError);
+      throws(() => d2.total({ unit: 'months' }), RangeError);
+      throws(() => d2.total({ unit: 'weeks' }), RangeError);
+    });
+    it('relativeTo is required for rounding durations with calendar units', () => {
+      throws(() => d.total({ unit: 'years' }), RangeError);
+      throws(() => d.total({ unit: 'months' }), RangeError);
+      throws(() => d.total({ unit: 'weeks' }), RangeError);
+      throws(() => d.total({ unit: 'days' }), RangeError);
+      throws(() => d.total({ unit: 'hours' }), RangeError);
+      throws(() => d.total({ unit: 'minutes' }), RangeError);
+      throws(() => d.total({ unit: 'seconds' }), RangeError);
+      throws(() => d.total({ unit: 'milliseconds' }), RangeError);
+      throws(() => d.total({ unit: 'microseconds' }), RangeError);
+      throws(() => d.total({ unit: 'nanoseconds' }), RangeError);
+    });
+    const d2Nanoseconds = d2.round({ largestUnit: 'nanoseconds', smallestUnit: 'nanoseconds' }).nanoseconds;
+    const totalD2 = {
+      days: d2Nanoseconds / (24 * 3.6e12),
+      hours: d2Nanoseconds / 3.6e12,
+      minutes: d2Nanoseconds / 6e10,
+      seconds: d2Nanoseconds / 1e9,
+      milliseconds: d2Nanoseconds / 1e6,
+      microseconds: d2Nanoseconds / 1e3,
+      nanoseconds: d2Nanoseconds
+    };
+    it('relativeTo is not required for rounding non-calendar units in durations without calendar units', () => {
+      equal(d2.total({ unit: 'days' }), totalD2.days);
+      equal(d2.total({ unit: 'hours' }), totalD2.hours);
+      equal(d2.total({ unit: 'minutes' }), totalD2.minutes);
+      equal(d2.total({ unit: 'seconds' }), totalD2.seconds);
+      equal(d2.total({ unit: 'milliseconds' }), totalD2.milliseconds);
+      equal(d2.total({ unit: 'microseconds' }), totalD2.microseconds);
+      equal(d2.total({ unit: 'nanoseconds' }), totalD2.nanoseconds);
+    });
+    const endpoint = relativeTo.add(d);
+    const options = (unit) => ({ largestUnit: unit, smallestUnit: unit, roundingMode: 'trunc' });
+    const fullYears = 6;
+    const fullDays = endpoint.since(relativeTo, options('days')).days;
+    const fullMilliseconds = endpoint.since(relativeTo, options('milliseconds')).milliseconds;
+    const partialDayMilliseconds = fullMilliseconds - fullDays * 24 * 3.6e6 + 0.005005;
+    const fractionalDay = partialDayMilliseconds / (24 * 3.6e6);
+    const partialYearDays = fullDays - fullYears * 365 + 2;
```

Comment:
Another good catch, thanks. Prettier is very opinionated about where you're allowed to put unnecessary parentheses, and after a while I found myself avoiding parentheses that Prettier might complain about. This might have been one of those cases. ;-(